src/Form/PrintArticleConvertForm.php

```diff
@@ -0,0 +1,194 @@
+<?php
+
+namespace Drupal\thunder_print\Form;
+
+use Drupal\Core\Entity\EntityTypeManagerInterface;
+use Drupal\Core\Form\FormBase;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\thunder_print\ArticleConverter;
+use Drupal\thunder_print\Entity\PrintArticleInterface;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * Class PrintArticleConvertForm.
+ *
+ * @package Drupal\thunder_print\Form
+ *
+ * @ingroup thunder_print
+ */
+class PrintArticleConvertForm extends FormBase {
+
+  /**
+   * The entity type manager service.
+   *
+   * @var \Drupal\Core\Entity\EntityTypeManagerInterface
+   */
+  protected $entityTypeManager;
+
+  protected $articleConverter;
+
+  /**
+   * PrintArticleSwitchTypeForm constructor.
+   *
+   * @param \Drupal\Core\Entity\EntityTypeManagerInterface $entityTypeManager
+   *   The entity type manager service.
+   * @param \Drupal\thunder_print\ArticleConverter $articleConverter
+   *   The articel converter service.
+   */
+  public function __construct(EntityTypeManagerInterface $entityTypeManager, ArticleConverter $articleConverter) {
+    $this->entityTypeManager = $entityTypeManager;
+    $this->articleConverter = $articleConverter;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container) {
+    return new static(
+      $container->get('entity_type.manager'),
+      $container->get('thunder_print.article_converter')
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getFormId() {
+    return 'PrintArticle_convertArticle';
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function buildForm(array $form, FormStateInterface $form_state, $print_article = NULL) {
+
+    /** @var \Drupal\thunder_print\Entity\PrintArticleInterface $print_article */
+    if ($print_article = $this->entityTypeManager->getStorage('print_article')
+      ->load($print_article)) {
+
+      $form['description']['#markup'] = $this->t('Select a target entity and bundle to convert the print article in. The current article will stay after that.');
+
+      $entity_types = [];
+
+      $targets = $this->getTargets($print_article);
+      foreach ($targets as $entity_type => $bundles) {
+        $definition = $this->entityTypeManager->getDefinition($entity_type);
+        $entity_types[$entity_type] = "{$definition->getLabel()} ({$definition->id()})";
+      }
+
+      $form['entity_type'] = [
+        '#type' => 'select',
+        '#title' => $this->t('Entity type'),
+        '#options' => $entity_types,
+        '#required' => TRUE,
+        '#ajax' => [
```

Comment:
When only one Entity type is configured for mapping, this should be preselected, the same should be done for the bundle selection below.
If bot, entity type and bundle, have only one selectable item, this whole step could be skipped.

Reply:
Furthermore, if no mappings are defined yet, please show a message, that a mapping is missing together with a link to the mapping forms.